ingestion/src/metadata/ingestion/models/custom_pydantic.py

```diff
@@ -30,13 +32,187 @@
 SECRET = "secret:"
 JSON_ENCODERS = "json_encoders"
 
+RESTRICTED_KEYWORDS = ["::", ">"]
+RESERVED_COLON_KEYWORD = "__reserved__colon__"
+RESERVED_ARROW_KEYWORD = "__reserved__arrow__"
+
+CREATE_ADJACENT_MODELS = ["ProfilerResponse", "SampleData"]
+
 
 class BaseModel(PydanticBaseModel):
     """
     Base model for OpenMetadata generated models.
     Specified as `--base-class BASE_CLASS` in the generator.
     """
 
+    @model_validator(mode="after")
+    @classmethod
+    def parse_name(  # pylint: disable=inconsistent-return-statements
+        cls, values
+    ):  # pylint: disable=too-many-locals
+        """
+        Primary entry point to process values based on their class.
+        Uses singledispatch to dynamically call appropriate processing functions.
+        """
+
+        from metadata.generated.schema.api.data.createDashboard import (
+            CreateDashboardRequest,
+        )
+        from metadata.generated.schema.api.data.createTable import CreateTableRequest
+        from metadata.generated.schema.api.data.createTableProfile import (
+            CreateTableProfileRequest,
+        )
+        from metadata.generated.schema.entity.data.table import (
+            Column,
+            Table,
+            TableConstraint,
+        )
+        from metadata.profiler.processor.core import ColumnProfile, ProfilerResponse
+        from metadata.profiler.source.metadata import ProfilerSourceAndEntity
+        from metadata.utils.entity_link import CustomColumnName
+
+        @singledispatch
+        def validate_name_and_transform(_):
```

Comment:
what's the benefit of having the validation logic inside the `BaseModel`? We could define the methods outside and call them from here right? Or is it about circular imports?